fabtests/ubertest/config.c

```diff
@@ -870,8 +870,7 @@ void fts_cur_info(struct ft_series *series, struct ft_info *info)
 	info->mode = (set->mode[series->cur_mode] == FT_MODE_NONE) ?
 			0 : set->mode[series->cur_mode];
 
-	if (set->msg_flags == FI_REMOTE_CQ_DATA)
-		info->mode |= FI_RX_CQ_DATA;
+	info->mode |= FI_RX_CQ_DATA;
```

Comment:
I understand why we can make this change. But if a test is using remote cq data, why is FI_REMOTE_CQ_DATA not set as a msg_flag?

Reply:
I'm more confused now.  If the test isn't using remote CQ data, why would it want to set this mode bit?

Reply:
I would need to update the config to set FI_REMOTE_CQ_DATA for tests that use it. But regardless, if there's some config that uses FI_RMA, it would need to support FI_RX_CQ_DATA otherwise verbs provider would return ENODATA.

Reply:
Providers can't unset mode bits.  Btw, I have a separate set of changes in this area that I'll repost later today.

Reply:
Perhaps I'm wrong... but I think that the rx_credits are only incremented when a completion is found and decremented when a buffer is posted. So on a writedata for example, if FI_RX_CQ_DATA is not set, the operation will generate a completion (rx_credits++) but the buffer won't get consumed so when we try to repost it, we get EAGAIN and never decrement the credits again, eventually causing assert(rx_credits <= max_credits) to fail.

Reply:
That's a good point. We should check the completion flags for FI_RECV before increasing rx_credits.